src/main/java/hu/dpc/phee/operator/importer/RecordParser.java

```diff
@@ -364,6 +355,28 @@ private void updateTransferTableForBatch(Long workflowInstanceKey, Long complete
             BatchFormatToTransferMapper.updateTransferUsingBatchDetails(transfer, batch);
             transferRepository.save(transfer);
         }
+    }
 
+    private void updateTransferTableWithFailedDuplicateTransaction(Long workflowInstanceKey, String filename) {
+        logger.info("Filename {}", filename);
+        if (filename == null) {
+            return;
+        }
+        filename = strip(filename);
+        String localFilePath = fileTransferService.downloadFile(filename, bucketName);
+        List<Transaction> transactionList = csvFileService.getTransactionList(localFilePath);
+        for (Transaction transaction : transactionList) {
+            Transfer transfer = BatchFormatToTransferMapper.mapToTransferEntity(transaction);
+            transfer.setStatus(TransferStatus.FAILED);
+            transfer.setBatchId(strip(tempDocumentStore.getBatchId(workflowInstanceKey)));
+            transfer.setStartedAt(new Date());
+            transfer.setCompletedAt(new Date());
+            transfer.setErrorInformation("Duplicate transaction");
```

Comment:
This reason can be obtained from file of this function is generic

Reply:
done, updated to `transfer.setErrorInformation(transaction.getNote());`.